framework/src/executioners/Transient.C

```diff
@@ -257,31 +257,35 @@ Transient::init()
   }
 
   _problem.initialSetup();
+
   _time_stepper->init();
 
   if (_app.isRestarting())
     _time_old = _time;
 
   _problem.outputStep(EXEC_INITIAL);
 
-  // If this is the first step
-  if (_t_step == 0)
-    _t_step = 1;
+  if (_app.isRecovering()) // Recover case
+  {
+    if (_t_step == 0)
+      mooseError("cannot at time step zero when recovering");
```

Comment:
When users get this error message, they will have no idea what is wrong and how to fix their inputs.

Reply:
It will probably be an internal error to trigger this. The problem is that I cannot think of any reason to hit this. It is here just for defensive purpose. It makes sure that when we shuffle codes, the condition here is still met and the code readers will know for sure at this point, `_t_step` must be zero. This check is not inside a loop, thus efficiency is not an issue. Do you have any better error message in mind?

Reply:
You could say:
```
Internal error in Transient executioner: _t_step equals 0 while recovering.
```